src/components/Loading.react.js

```diff
@@ -37,9 +37,7 @@ export default class Loading extends Component {
             type,
         } = this.props;
 
-        const isLoading = getLoadingState(this);
-
-        if (isLoading) {
+        if (loading_state && loading_state.is_loading) {
```

Comment:
The nested loading state is already determined by the renderer. Only need to check the prop value